packages/framework/store/src/store/doc/block-collection.ts

```diff
@@ -63,11 +61,58 @@ export class BlockCollection extends Space<FlatBlockMap> {
 
   private readonly _idGenerator: IdGenerator;
 
+  private _initSubDoc = () => {
+    let subDoc = this.rootDoc.spaces.get(this.id);
+    if (!subDoc) {
+      subDoc = new Y.Doc({
+        guid: this.id,
+      });
+      this.rootDoc.spaces.set(this.id, subDoc);
+      this._loaded = true;
+      this._onLoadSlot.emit();
+    } else {
+      this._loaded = false;
+      this.rootDoc.on('subdocs', this._onSubdocEvent);
+    }
+
+    return subDoc;
+  };
+
+  private _loaded!: boolean;
+
+  private _onLoadSlot = new Slot();
+
+  private _onSubdocEvent = ({ loaded }: { loaded: Set<Y.Doc> }): void => {
+    const result = Array.from(loaded).find(
+      doc => doc.guid === this._ySpaceDoc.guid
+    );
+    if (!result) {
+      return;
+    }
+    this.rootDoc.off('subdocs', this._onSubdocEvent);
+    this._loaded = true;
+    this._onLoadSlot.emit();
+  };
+
   /** Indicate whether the block tree is ready */
   private _ready = false;
 
   private _shouldTransact = true;
 
+  protected readonly _yBlocks: Y.Map<YBlock>;
+
+  /**
+   * @internal Used for convenient access to the underlying Yjs map,
+   * can be used interchangeably with ySpace
+   */
+  protected readonly _ySpaceDoc: Y.Doc;
```

Comment:
After removing the space, some names can also be improved.

Reply:
I'll update them and use signal in next pr.